src/network/network_server.cpp

```diff
@@ -207,12 +207,8 @@ struct PacketWriter : SaveFilter {
  * Create a new socket for the server side of the game connection.
  * @param s The socket to connect with.
  */
-ServerNetworkGameSocketHandler::ServerNetworkGameSocketHandler(SOCKET s) : NetworkGameSocketHandler(s)
+ServerNetworkGameSocketHandler::ServerNetworkGameSocketHandler(SOCKET s, NetworkAddress client_address) : NetworkGameSocketHandler(s, _network_client_id++), status(STATUS_INACTIVE), receive_limit(_settings_client.network.bytes_per_frame_burst), client_address(client_address)
```

Comment:
Lol, two years later, same comment: please don't do a ++ in a ctor initializer like this. That is going to be overlooked and will bite us in the ass :) ctor member initializer should be without side-effects. Side-effects go in the body.